robot/impl/local_robot.go

```diff
@@ -1096,6 +1100,16 @@ func dialRobotClient(
 // a best effort to remove no longer in use parts, but if it fails to do so, they could
```

Comment:
The line above is off-diff. But it says:
> // Reconfigure will safely reconfigure

This method is no longer the "safe" version. I would have expected a more conservative PR would be to have `Reconfigure` be `forceSync=true` and introduce `ReconfigureAsync` being `forceSync=false`.

That way we'd be certain that tests aren't adopting an algorithm that we've yet to convince ourselves safe for all situations.

Reply:
that's fair, we could merge this feature darkly (with a flag perhaps?) and focus on testing. I was planning on expanding [this test matrix](https://github.com/viamrobotics/rdk/pull/3707/files#r1600236702) along with manual testing to prove out our new algorithm, but I can continue that work post-merge.

also for what it's worth, the "safe" in this code comment doesn't really mean anything tangible IMO

Reply:
I feel pretty good about enabling this code for cases where all of the resources are new - it's just a matter of detecting that properly.